test/install_test.go

```diff
@@ -101,18 +111,24 @@ func TestCheckPreInstall(t *testing.T) {
 	}
 }
 
-func TestInstall(t *testing.T) {
-	cmd := []string{"install",
-		"--controller-log-level", "debug",
-		"--proxy-log-level", "warn,linkerd2_proxy=debug",
-		"--linkerd-version", TestHelper.GetVersion(),
-	}
-	if TestHelper.AutoInject() {
-		cmd = append(cmd, []string{"--proxy-auto-inject"}...)
+func TestInstallOrUpgrade(t *testing.T) {
+	var (
+		cmd  = "install"
+		args = []string{
+			"--controller-log-level", "debug",
+			"--proxy-log-level", "warn,linkerd2_proxy=debug",
+			"--linkerd-version", TestHelper.GetVersion(),
+		}
+	)
+
+	if TestHelper.UpgradeFrom() != "" {
+		cmd = "upgrade"
+		args = append(args, "--proxy-auto-inject")
```

Comment:
related to earlier comment, best to decouple `TestHelper.UpgradeFrom()` from `--proxy-auto-inject`